testutil-server/src/test/java/io/spine/testing/server/EmittedMessageSubjectTest.java

```diff
@@ -101,15 +112,59 @@ void retrieveMessage() {
     @DisplayName("check the message count when obtaining a ProtoSubject")
     void failOnIndexOutOfBounds() {
         int messageCount = 5;
-        Iterable<M> messages = messages(messageCount);
+        Iterable<W> messages = messages(messageCount);
         int index = 13;
         @SuppressWarnings("CheckReturnValue")
         AssertionError error = expectFailure(whenTesting -> whenTesting.that(messages)
                                                                        .message(index));
         TruthFailureSubject assertError = assertThat(error);
-        assertError.factValue(MESSAGE_COUNT_FACT_KEY)
+        assertError.factValue(FactKey.MESSAGE_COUNT.value())
                    .isEqualTo(String.valueOf(messageCount));
-        assertError.factValue(REQUESTED_INDEX_FACT_KEY)
+        assertError.factValue(FactKey.REQUESTED_INDEX.value())
                    .isEqualTo(String.valueOf(index));
     }
+
+    @Test
+    @DisplayName("obtain derived subject with messages of type")
+    void subSubjectByType() {
+        int messageCount = 3;
+        int otherMessageCount = 2;
+        Iterable<W> outerObjects = ImmutableList
+                .<W>builder()
+                .addAll(messages(messageCount))
+                .addAll(messages(otherMessageCount, this::createAnotherMessage))
+                .build();
+
+        Class<M> cls1 = typeOf(createMessage());
+        Class<M> cls2 = typeOf(createAnotherMessage());
```

Comment:
Please use `message`  and `other` instead of `1` and `2` for better readability.